src/io/syncsocket.c

```diff
@@ -190,7 +190,7 @@ MVMint64 socket_write_bytes(MVMThreadContext *tc, MVMOSHandle *h, char *buf, MVM
     while (bytes > 0) {
         int r;
         do {
-            r = send(data->handle, buf, (int)bytes, 0);
+            r = send(data->handle, buf, bytes > 32767 ? 32767 : bytes, 0);
```

Comment:
I think that this should be 1073741824 too. The `send` call should fail if it what is passed is too large.

In that, the IBM document mentions TCP, UDP and RAW sockets. For UDP and RAW sockets, it matters whether the bytes are transmitted atomically (ie one packet, possibly fragmented at the IP level). The remote end is going to be reading in messages, not streams. Hence what is being sent is too big, it's better for the sender to get the error `EMSGSIZE` (and know this) than send a partial packet (pretend nothing happened), (which the receiver might not realise is truncated) and then send a second (or third) packet which starts midway through the intended message, and might confuse the receiver.

Thinking further, this code is actually incorrect (assuming you can reach it with a UDP socket), as it prevents you sending a UDP packet sized between 32767 and 65527 bytes, which is a legitimate use.

So I think that this should be 1073741824 too. (And it might be better to express that as `0x40000000` in all places - sorry, didn't think of that earlier). Trying to send large over-large payloads with UDP and RAW will fail successfully, whilst large transfers with TCP potentially will use much larger packets. (I think that it's possible for TCP to use packets > 65536ish bytes if the underlying link supports them)(yes there are buffers, but network cards can be really fast, so they might have sent your first 32767 bytes before you manage to enqueue another 32767 in the buffer)

Disclaimer - as best I can remember I've never actually written code to deal with UDP or RAW sockets. I remember playing with ["ping of death"](https://en.wikipedia.org/wiki/Ping_of_death) at the time, but IIRC there was other peoples' source code, and heck, Irix `ping` was trusting enough to send such malformed packets if you simply gave it daft comand line flags - the "feature" was standard on the OS.

Reply:
Done, still passes all tests and `raku -e 'say "H" x 2147483647;' | wc` reports the correct value.